src/lib/Types/Core.hs

```diff
@@ -235,6 +234,12 @@ instance GenericB EffectBinder where
   fromB (EffectBinder effs) = LiftB effs
   toB   (LiftB effs) = EffectBinder effs
 
+data BoxPtr n = BoxPtr (Atom n) (Block n)  -- ptrptr, size
+                deriving (Show, Generic)
+
+data NonDepNest ann n l = NonDepNest (Nest AtomNameBinder n l) [ann n]
```

Comment:
What here is not supposed to depend on what?  Could you add a code comment?

Reply:
Will do. In a `Nest b n l` the free vars of a binder half way through the nest can depend on all binders above. In a `NonDepNest` the annotations of each binder cannot depend on the names bound previously. You can think of it as a parallel binding construct, that introduces multiple names into the scope concurrently. But safer names always reasons about linear scope extensions, so we encode it in this way instead.